litellm/proxy/proxy_server.py

```diff
@@ -3064,7 +3073,7 @@ async def completion(
         elif (
             llm_router is not None and data["model"] in router_model_names
         ):  # model in router model list
-            response = await llm_router.atext_completion(**data)
+            response = llm_router.text_completion(**data)
```

Comment:
@eslamkarim making an async call sync would really impact perf. 

The text completion issue you were seeing should've been addressed recently. If not, happy to address that in a separate issue.